Lesson16/duygu/script.js

```diff
@@ -0,0 +1,356 @@
+      - Log the resulting array.
+  */
+  function filterLongWords(words, minLength) {
+    const filteredWords = [];
+    for (const word of words) {
+      if (word.length >= minLength) {
+        filteredWords.push(word);
+      }
+    }
+    return filteredWords;
+  }
+  
+  console.log(filterLongWords(["apple", "banana", "pear", "grape", "strawberry"], 5));
+  console.log(filterLongWords(["cat", "dog", "elephant", "rat"], 4));
+  
+  
+  /*
+  16. Log Array Elements with Their Indices
+      - Define a function `logElementsWithIndex(arr)` that loops through the array
+        and logs "Index: i, Value: arr[i]" for each element.
+  */
+  function logElementsWithIndex(arr) {
+    for (let i = 0; i < arr.length; i++) {
+      console.log(`Index: ${i}, Value: ${arr[i]}`);
+    }
+  }
+  
+  logElementsWithIndex(["apple", "banana", "cherry"]);
+  logElementsWithIndex([10, 20, 30, 40]);
+  
+  /*
+  17. Find the Smallest Number in an Array
+      - Define a function `findMin(numbers)` that loops through the array
+        to find and return the smallest number.
+      - Log the smallest number.
+  */
+  function findMin(numbers) {
+    let min = numbers[0];
+    let i = 1;
+    while (i < numbers.length) {
+      if (numbers[i] < min) {
+        min = numbers[i];
+      }
+      i++;
+    }
+    return min;
+  }
+  
+  console.log(findMin([10, 5, 8, 3, 7]));
+  console.log(findMin([15, 20, -5, 0, 30]));
+  
+  /*
+  18. Count Occurrences of a Word in an Array
+      - Define a function `countOccurrences(arr, word)` that loops through `arr`
+        to count how many times `word` appears.
+      - Log the count.
+  */
+  function countOccurrences(arr, word) {
+    let count = 0;
+    for (let i = 0; i < arr.length; i++) {
+      if (arr[i] === word) {
+        count++;
+      }
+    }
+    return count;
+  }
+  
+  console.log(countOccurrences(["apple", "banana", "apple", "cherry", "apple"], "apple"));
+  console.log(countOccurrences(["cat", "dog", "cat", "mouse", "cat"], "cat"));
+  console.log(countOccurrences(["red", "blue", "green"], "yellow"));
+  
+  /*
+  19. Remove Falsy Values
+      - Define a function `removeFalsyValues(arr)` that loops through an array
+        and returns a new array without falsy values (false, 0, "", null, undefined, NaN).
+      - Log the new array.
+  */
+  function removeFalsyValues(arr) {
+    const result = [];
+    for (const item of arr) {
+      if (item) {
+        result.push(item);
+      }
+    }
+    return result;
+  }
+  
+  console.log(removeFalsyValues([0, 1, false, 2, "", 3, null, undefined, NaN, 4]));
+  console.log(removeFalsyValues(["apple", "", "banana", null, "cherry", false]));
+  
+  /*
+  20. Sum of All Digits in a String
+      - Define a function `sumDigits(str)` that loops through each character of `str`,
+        checks if it's a digit, and if so, adds it to a total sum.
+      - Log the final sum.
+      - Example: "abc123" -> 6
+  */
+  
+  
```

Comment:
`!isNaN(char)` is used to check if a character is non a number, then use the opposite result (as we want numbers only).
`Number(char)` converts the char to a number.

```suggestion
  function sumDigits(str) {
    let sum = 0;
    for (const char of str) {
        if (!isNaN(char) && char !== ' ') {
            sum += Number(char);
        }
    }
    console.log(sum);
    return sum;
}
```